scripts/openssl.sh

```diff
@@ -48,7 +48,9 @@ rm -rf "openssl-$version"
 mv "openssl-$tag" "openssl-$version"
 
 cd "openssl-$version"
-./config $config
+# -d makes a debug build which helps with debugging memory issues and
+# other problems. It's not necessary for normal use.
+./config -d $config
```

Comment:
I wonder if this should be done as an optional flag on the script in case debug mode changes behavior, ending up making us miss some edge case. On the other hand, I think this might enable debug assertions and let us catch other issues earlier (e.g. we violate some invariant).

Or would it be justified to add another variable to the build matrix and run both ways?

Reply:
The chances to miss an edge-case by using a debug build are very low, but they do exist. Yet, I don't think it matters in practice as downstream users won't be using the OpenSSL libraries we use for testing, either because the patch version is not the same, or because they patch OpenSSL, or because the build configuration is different.

With this in mind, I think that it's up to the Microsoft and RedHat Go forks to test on environments that would match better their end users.